File: lwjgl/src/main/java/com/darkraha/opengldemoj/gl/LightDirectional.java
package com.darkraha.opengldemoj.gl;

import org.joml.Vector3f;

/**
 * For simple directional light
 */
public class LightDirectional {

    public LightDirectional() {

    }

    public LightDirectional(Vector3f ambient, Vector3f diffuseColor, Vector3f direction) {
        this.ambient = ambient;
        this.diffuseColor = diffuseColor;
        this.direction = direction;
    }

    public Vector3f ambient;
    public Vector3f diffuseColor;
    public Vector3f direction;

}

File: android/app/src/main/java/com/darkraha/gldemos/gl/LightDirectional.kt
package com.darkraha.gldemos.gl

import org.joml.Vector3f

class LightDirectional {
    constructor(ambient: Vector3f?, diffuseColor: Vector3f?, direction: Vector3f?) {
        this.ambient = ambient
        this.diffuseColor = diffuseColor
        this.direction = direction
    }

    constructor() : this(null, null, null)

    var ambient: Vector3f? = null
    var diffuseColor: Vector3f? = null
    var direction: Vector3f? = null
}